### PR TITLE
Fix exceptional weither icon size (#6425)

### LINKED ISSUE #6425: Weather card exceptional icon not sized correctly
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [ ] I have tried a different browser to see if it is related to my browser.

## The problem

The exception icon is too large compared to other icons.  It also obscures the temperature.  Location redacted below.

Screenshot_20200717-215610~2

## Expected behavior

Icon to be of similar size and not obscure temperature.

## Steps to reproduce

Have a weather entity with an exceptional state in forecast and display it in weather card.

## Environment

- Home Assistant release with the issue: 0.112.5
- Last working Home Assistant release (if known):
- Browser and browser version: Chrome Android
- Operating system: Android

## State of relevant entities

A forecast entry with (I cannot copy from the states page from chrome browser on Android, but this is a different issue).
```yaml
condition: exceptional
```

## Problem-relevant configuration

UI configured NWS integration

## Javascript errors shown in your browser console/inspector

```txt

```

### PR DESCRIPTION
**Pull request by gibwar**
The weather icon for the `exceptional` state uses a different DOM layout
than the other icons, using a `.weather-icon` class that sets the size
to 64px (52px on narrow). This updates the `forecast-icon > *` class to
set the correct variable to match the expected `40px` and works on all
sizes from `veryverynarrow` to normal.

This does remove the sizes for the `.weather-icon` class, but seeing that
this is a self-contained card, I don't think there is any impact outside
this change.

## Proposed change
This fixes the exceptional icon size to match the 40px of the other icons.
### Issue
issue

### Fix
fix
(the above screenshot shows they veryverynarrow, verynarrow, narrow, and normal layouts, in that order)

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
Configure NWS integration from GUI. Note, since this uses live weather data,
I overrode `/src/panels/lovelace/cards/hui-weather-forecast-card.ts` line 257
to read:
```typescript
${getWeatherStateIcon('exceptional', null)
```
to get the screenshots above.

## Additional information

- This PR fixes or closes issue: fixes #6425
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @gibwar, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#6634).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/panels/lovelace/cards/hui-weather-forecast-card.ts [hunk 1]
```diff
@@ -353,10 +353,6 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
           height: 64px;
         }
 
-        .weather-icon {
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-weather-forecast-card.ts:356]: Why was this removed?

Comment (zsarnett) [reply src/panels/lovelace/cards/hui-weather-forecast-card.ts:356]: What if the exception icon is the main icon and not in the forecast

File: src/panels/lovelace/cards/hui-weather-forecast-card.ts [hunk 2]
```diff
@@ -469,10 +466,6 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
           width: 52px;
         }
 
-        :host([narrow]) .weather-icon {
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-weather-forecast-card.ts:472]: Same here

File: src/panels/lovelace/cards/hui-weather-forecast-card.ts (hunk 1 above)
Comment (gibwar) [reply src/panels/lovelace/cards/hui-weather-forecast-card.ts:356]: Ah, I see how it can be used up there. Give me a few and I'll push a fix.

Comment (gibwar) [reply src/panels/lovelace/cards/hui-weather-forecast-card.ts]: Fixed.

File: src/panels/lovelace/cards/hui-weather-forecast-card.ts (hunk 2 above)
Comment (gibwar) [reply src/panels/lovelace/cards/hui-weather-forecast-card.ts:472]: Fixed by restricting the selector to `:host([narrow]) .icon-image .weather-icon` so it only applies if it's the main icon and not a forecast icon.

**Comment by gibwar** [comment]
Updated the PR resolving @zsarnett's comments. I missed that `.weather-icon` was also used for the main icon as well as the exceptional icon. Resolved by making sure that the 40px applies to forecast icons and restricted the main image to apply to ` .icon-image`.

**Review by zsarnett — APPROVED** [review]

**Pull request merged**